File: backend/app/schemas/student_directory.py
from pydantic import BaseModel, EmailStr, ConfigDict
from typing import Optional, List, Any
from datetime import datetime

class DirectoryStudentCreate(BaseModel):
    name: str
    email: Optional[str] = None
    roll_number: Optional[str] = None
    phone: Optional[str] = None
    student_code: Optional[str] = None
    status: Optional[str] = "active"

class DirectoryStudentUpdate(BaseModel):
    name: Optional[str] = None
    email: Optional[str] = None
    roll_number: Optional[str] = None
    phone: Optional[str] = None
    status: Optional[str] = None

class DirectoryStudentResponse(BaseModel):
    id: str
    directory_id: str
    name: str
    email: Optional[str] = None
    roll_number: Optional[str] = None
    phone: Optional[str] = None
    student_code: Optional[str] = None
    status: str
    created_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)

class StudentDirectoryCreate(BaseModel):
    name: str
    description: Optional[str] = None
    initial_students: Optional[List[DirectoryStudentCreate]] = None

class StudentDirectoryUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None

class StudentDirectoryResponse(BaseModel):
    id: str
    workspace_id: str
    name: str
    description: Optional[str] = None
    created_by: str
    student_count: Optional[int] = 0
    is_active: bool
    created_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)

class CSVImportResult(BaseModel):
    imported_count: int
    skipped_count: int
    errors: List[str]

File: frontend/types/studentDirectory.ts
export interface DirectoryStudent {
  id: string;
  directory_id: string;
  name: string;
  email: string;
  roll_number?: string | null;
  phone?: string | null;
  status: 'active' | 'inactive';
  created_at: string;
}

export interface StudentDirectory {
  id: string;
  workspace_id: string;
  name: string;
  description?: string | null;
  created_by?: string | null;
  is_active: boolean;
  student_count: number;
  created_at: string;
  updated_at: string;
}

export interface DirectoryStudentCreate {
  name: string;
  email: string;
  roll_number?: string;
  phone?: string;
}

export interface StudentDirectoryCreate {
  name: string;
  description?: string;
  initial_students?: DirectoryStudentCreate[];
}

export interface CSVImportError {
  row: number;
  email?: string;
  reason: string;
}

export interface CSVImportResult {
  total_rows: number;
  imported_count: number;
  skipped_count: number;
  errors: CSVImportError[];
}
